### ISSUE TITLE
Evaluate SQS for basket celery broker

### ISSUE THREAD

**Issue by pmac** [mozmeao/infra#1149]
Primary questions:

1. Is SQS a more secure solution for data in transit on the celery queue
2. Is SQS a cheaper solution for basket's celery queue

[SQS celery broker docs](http://docs.celeryproject.org/en/latest/getting-started/brokers/sqs.html)

**Comment by pmac** [comment]
So far I think this would be a great idea using the FIFO queues. It appears as though it would be nearly free for basket's usage volume, supports data encryption, and should be plenty fast enough. Only question remaining for me is if basket moving away from Redis would really allow us to scale down our use of Redis or not?

**Comment by pmac** [comment]
More questions:

* Would this move make it more difficult to move from AWS to something like GKE?
* We would probably need a memcache instance if no redis is available, would that negate the cost savings?

**Comment by pmac** [comment]
thoughts @mozmeao/theconn ?

**Comment by metadave** [comment]
> Would this move make it more difficult to move from AWS to something like GKE?

Probably, we'd have to find a similar service in GCP (maybe Pub/Sub?). We might have to switch back to Redis, and run it in-cluster.

> We would probably need a memcache instance if no redis is available, would that negate the cost savings?

We might be ok wrt reserved instances, but yeah, seems like it could negate cost savings. We can try to put together some numbers from: https://calculator.s3.amazonaws.com/index.html 
Happy to help with that, I've used that tool a bit too much.

**Comment by pmac** [comment]
I don't think basket absolutely requires a shared cache, but it does help with SFDC auth and some other caching over just local memory caching. If it's too expensive it's probably not worth it though.

**Comment by jgmize** [comment]
https://medium.com/@Twistacz/whats-wrong-with-sqs-and-why-we-gave-it-up-22780d4f48ef

**Comment by pmac** [comment]
2 things:

1. this article is from 2 years ago
2. we don't use a results queue or the celery monitoring stuff and I think reduced cloud watch is fine (assuming that's even still the case).

I did see in the docs that these features aren't supported for SQS but we also don't use them. Still seems worth a shot. I filed https://github.com/mozmeao/basket/pull/296 to get basket ready to use SQS if we want to try it.
